Architecture.md

```diff
@@ -73,10 +101,51 @@
 
 #### Client Applications
 
-***TODO: [INFRA] how are the Client applications provisioned, how they are exposed and how the underlying infrastructure is managed***
+Pomegranate is responsible for starting, stopping, and interacting with the client applications.
+It is also responsible for managing the networking of the applications.
+
+Pomegranate is completely stateless, and the API is the only external way to interact with it.
+Once a container is started, Pomegranate does not keep track of it, and it is the responsibility of the execution engine to manage it.
+
+```mermaid
+flowchart TD
+    web([port 80])
+    webscure([port 443])
+    traefik[Traefik]
+    app_1[User app 1]
+    app_2[User app 2]
+    app_3[User app 3]
+
+    rules{Resolve\nvia labels}
+
+    webscure --> |HTTPS| traefik
+    web --> |HTTP| traefik
+    traefik --> rules
+
+    rules --> |**uuid_1**.user-app:80| app_1
+    rules --> |**uuid_2**.user-app:80| app_2
+    rules --> |**uuid_3**.user-app:80| app_3
+
+```
+The [FQDN (Fully Qualified Domain Name)](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) is passed to Pomegranate via an environment variable.
```

Comment:
Avoid directly saying "FQDN", prefer "domain name". The environment variable is not even an FQDN (the domain name you pass doesn't end with a `.` character).

Also, by reading this, I don't know what domain name you refer to nor how it is being used.

Reply:
rip